packages/cli/src/commands/login/index.ts

```diff
@@ -64,63 +39,25 @@ export default async function login(client: Client): Promise<number> {
     return 2;
   }
 
-  const input = parsedArgs.args[1];
-
-  let result: LoginResult = 1;
+  const obsoleteFlags = Object.keys(parsedArgs.flags);
 
-  if (input) {
-    // Email or Team slug was provided via command line
-    if (validateEmail(input)) {
-      result = await doEmailLogin(client, input);
-    } else {
-      result = await doSamlLogin(client, input, parsedArgs.flags['--oob']);
-    }
-  } else if (parsedArgs.flags['--github']) {
-    result = await doGithubLogin(client, parsedArgs.flags['--oob']);
-  } else if (parsedArgs.flags['--gitlab']) {
-    result = await doGitlabLogin(client, parsedArgs.flags['--oob']);
-  } else if (parsedArgs.flags['--bitbucket']) {
-    result = await doBitbucketLogin(client, parsedArgs.flags['--oob']);
-  } else {
-    // Interactive mode
-    result = await prompt(client, undefined, parsedArgs.flags['--oob']);
+  if (obsoleteFlags.length) {
+    const flags = obsoleteFlags.map(f => chalk.bold(f)).join(', ');
+    output.warn(`The following flags are deprecated: ${flags}`);
   }
 
-  // The login function failed, so it returned an exit code
-  if (typeof result === 'number') {
-    return result;
+  const obsoleteArguments = parsedArgs.args.slice(1);
+  if (obsoleteArguments.length) {
+    const args = obsoleteArguments.map(a => chalk.bold(a)).join(', ');
+    output.warn(`The following arguments are deprecated: ${args}`);
   }
 
-  // Save the user's authentication token to the configuration file.
-  client.authConfig.token = result.token;
-
-  if (result.teamId) {
-    client.config.currentTeam = result.teamId;
-  } else {
-    delete client.config.currentTeam;
+  if (obsoleteArguments.length || obsoleteFlags.length) {
+    output.print(
+      // TODO: fix link
+      `Read more in our ${output.link('changelog', 'https://vercel.com/changelog')}.\n`
+    );
   }
 
-  // If we have a brand new login, update `currentTeam`
-  if (isInitialLogin) {
-    await updateCurrentTeamAfterLogin(client, client.config.currentTeam);
-  }
-
-  writeToAuthConfigFile(client.authConfig);
-  writeToConfigFile(client.config);
-
-  output.debug(`Saved credentials in "${hp(getGlobalPathConfig())}"`);
-
-  output.print(
-    `${chalk.cyan('Congratulations!')} ` +
-      `You are now logged in. In order to deploy something, run ${getCommandName()}.\n`
-  );
-
-  output.print(
-    `${prependEmoji(
-      `Connect your Git Repositories to deploy every branch push automatically (https://vercel.link/git).`,
-      emoji('tip')
-    )}\n`
-  );
-
-  return 0;
+  return await future(client);
```

Comment:
Nit: it will no longer be the "future" method, just _the_ login method. Can we rename it and the file?